Budget App

HansZizold opened the pull request:
### BUDGET APP

Here I present the Budget App project developed according to the requirements, which I resume:

- The project views were developed following the design guidelines provided.

- INTERACTIONS
  - A Splash screen to start.
  - A login page that allows the user to sign up and log in.
  - A categories page with all the categories the user has.
  - A transaction page for each category.
  - Forms to add new categories and new transactions.

  [Video presenting the features of the project](https://www.loom.com/share/9386403b06c34c8abea3a7beaa11c11b)

- TESTING
  - Unit and integration tests for all the most important components were developed.
 
- TECHNICAL FEATURES 
  - Postgres was used for the database.
  - Devise was used for authentication.
  - Tailwind was used for styling.

  [Complete Requirements](https://github.com/microverseinc/curriculum-rails/blob/main/capstone/rails_capstone.md)

Happy review!!!

HansZizold committed:
First Two Days

Models
Splash Page
Categories Page

HansZizold committed:
Readme File

HansZizold committed:
Tests

HansZizold committed:
More tests

HansZizold committed:
Tests Finished

HansZizold committed:
Project Finished

HansZizold committed:
Update README.md

rustam-tolipov reviewed (commented):
Hi @HansZizold, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that needed to include a future features section in the readme.

Please note that all projects must include a Future Features section outlining 1-3 potential features to be added. You have submitted your project without this section, so please add it before submitting again. Thank you for your understanding.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

HansZizold committed:
Update Readme file

HansZizold commented:
Hi @Rustamxon7, thank you for your observation.

I added some future features that I will add to the project.

Regards,

elisha2kyakpo1 commented on README.md:
```diff
@@ -1,2 +1,214 @@
+<!-- LICENSE -->
+
+## 📝 License <a name="license"></a>
+
+This project is [MIT](./LICENSE.md) licensed.
+
```
- [ ] Kindly fix the license link, currently, I am getting a 404 error.

rotimiazeez replied:
I'm still getting the 404 error when I click on the MIT License link 😅. Kindly fix this 👍🏼

elisha2kyakpo1 commented on db/migrate/20230213204306_create_groups.rb:
```diff
@@ -0,0 +1,11 @@
+  def change
+    create_table :groups do |t|
+      t.string :name
+      t.string :icon
+      t.references :user, null: false, foreign_key: true
```
- [ ] Kindly change the name of a user's foreign entity to read `author_id`. To do this, you can do something like this: `t.references :author, foreign_key: {to_table: :users}`

elisha2kyakpo1 commented on db/migrate/20230213204354_create_operations.rb:
```diff
@@ -0,0 +1,11 @@
+    create_table :operations do |t|
+      t.string :name
+      t.float :amount
+      t.references :user, null: false, foreign_key: true
+
```
- [ ] Kindly change the name of a user's foreign entity to read `author_id`. To do this, you can do something like this: `t.references :author, foreign_key: {to_table: :users}`

![author_id](https://user-images.githubusercontent.com/53124548/219878169-57e1693d-070d-43e0-aeb4-755b366cf3ea.png)

elisha2kyakpo1 commented on spec/model/user_spec.rb:
```diff
@@ -0,0 +1,21 @@
+RSpec.describe User, type: :model do
+  user = User.new(name: 'John', surname: 'Doe', email: 'john@test.com', password: '123456')
+  user.confirm
+  before { user.save }
```
- [ ] I got an error when I tried to run the test command, the error was coming from line 5. Kindly fix the error or make sure that how to test the project is clearly stated in the readme file so that the next reviewer will test it without any problems.

![confirm](https://user-images.githubusercontent.com/53124548/219878131-10ad6190-6f16-4863-93a1-182d9bf8943d.png)

elisha2kyakpo1 commented on app/views/groups/index.html.erb:
```diff
@@ -0,0 +1,33 @@
+<div class = "flex flex-row justify-start w-full bg-[#3778c2] text-white text-center py-3 text-xl">
+  <%= button_to "EXIT", destroy_user_session_path, method: :delete, class: 'ml-4 text-[12px]' %>
+  <h2 class = "w-full mr-4" >CATEGORIES</h2>
+</div>
+
+<div>
+  <div class = "flex flex-col bg-[#f1f3f2] w-full">
```
- [ ] When we examine the given design, you will notice a hamburger menu containing user details such as name. kindly add the hamburger menu to meet the project requirements

elisha2kyakpo1 reviewed (changes requested):
Hi 👋 @HansZizold, 

Good job so far!
You still need to work on some issues to go to the next project, but you are almost there!

## Highlights

✔️ The PR title and description look good and professional.
✔️ No linter errors.
✔️ Descriptive readme file.
✔️ The app works as expected👏

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

HansZizold committed:
Tests and License Corrections

HansZizold commented:
Hi @elisha2kyakpo1, 

Thank you very much for your comments.
I corrected the test part and added the license file.

Change the name of a user's foreign entity by author_id: you're right, it is in the diagram. But I request you consider this observation since this change doesn't impact the user experience and doesn't impact the project itself. I would do it, but the time I have is not enough to do this change and all the changes that I would have to make in all the logic and tests of the project. 

Hamburger button: you're right, it is in the design, but it was not clear to me that this button was necessary for the project. There are also other cards in the design that does not apply to the project, so I asked other people that got their projects approved and they told me that this button was not necessary. For this reason, I ask you to reconsider this observation.

Regards,

Hans

rotimiazeez reviewed (changes requested):
Hi @HansZizold 😉, 

Great work on making the changes 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

## Highlight
- All test are now passing 👌🏽

## Suggested changes
_Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me(@rotimiazeez) in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid, unless it is requested otherwise._**

elisha2kyakpo1 commented:
Hi @HansZizold, I'm sorry that I didn't see your message in time, you are right about the hamburger menu, but the requirements state that a name of a user should be on the home page. In this case, you can just add a user's name on the home page.

HansZizold committed:
License file

Adding missing extension to the file

HansZizold commented:
Hi @rotimiazeez,

Very sorry for the License file. The extension was missing. I corrected it.

Regards,

m05e5 reviewed (approved):
**Hi @HansZizold 
You did great. Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
![CleverScornfulBuffalo-size_restricted](https://user-images.githubusercontent.com/37901469/208869296-5ca9f6cc-a1b5-48fd-bd59-78fd83b35471.gif)

## Highlights

✔️ Nice UI
✔️ No linters error
✔️ Clean and well-organized code
✔️ Excellent communication with reviewers

## Optional suggestions



_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

_**m05e5**_

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._**
